salt/_modules/caasp_hosts.py

```diff
@@ -150,20 +162,25 @@ def _add_names(hosts, ips, names):
     for ip in ips:
         __utils__['caasp_log.debug']('hosts: adding %s -> %s', ip, names)
         if ip not in hosts:
-            hosts[ip] = _concat([], names)
+            hosts[ip] = insert_fun([], names)
         else:
-            hosts[ip] = _concat(hosts[ip], names)
+            hosts[ip] = insert_fun(hosts[ip], names)
 
 
-def _add_names_for(hosts, nodes_dict, infra_domain):
+def _add_names_for(hosts, nodes_dict, infra_domain, insert_fun=_sorted_append):
     for id, ifaces in nodes_dict.items():
         ip = __salt__['caasp_net.get_primary_ip'](host=id, ifaces=ifaces)
         if ip:
-            _add_names(hosts, ip, [id, id + '.' + infra_domain])
+            _add_names(hosts, ip, [id, id + '.' + infra_domain], insert_fun)
 
+
+def _add_nodenames_for(hosts, nodes_dict, infra_domain):
+    for id, ifaces in nodes_dict.items():
+        ip = __salt__['caasp_net.get_primary_ip'](host=id, ifaces=ifaces)
+        if ip:
             nodename = __salt__['caasp_net.get_nodename'](host=id)
```

Comment:
We could use guard clause here for simplifying, if not ip return nothing then continue (pseudo code)

Reply:
I've never been really fond of those guard clauses. I prefer a style where you can clearly see that you are going deep down the rabit hole... but it is my personal taste.

Reply:
yop i think is personal taste. To me guard clause simplify the `context` infos and nested things and you see basically when you go down the func you follow  the main logic, otherwise you exit with guardClause

Reply:
Let's leave as it is right now, following the style of other parts of the file. If we change this to a guard clause other parts of this file would need adapting as well and it's out of the scope of this PR.